docs/design/2022-07-20-insert-support.md

```diff
@@ -0,0 +1,127 @@
+# Support Insert SQL 
+
+- Author(s): TrafalgarRicardoLu
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2462
+
+## Table of Contents
+- [Introduction](#introduction)
+- [Motivation or Background](#motivation-or-background)
+- [Goals](#goals)
+- [API](#api)
+- [Detailed Design](#detailed-design)
+  * [Add new capability](#add-new-capability)
+  * [Remove write options](#remove-write-options)
+  * [Implement V2 interface using V1 code](#implement-v2-interface-using-v1-code)
+  * [Compatible with different Spark versions](#compatible-with-different-spark-versions)
+  * [Why not V2 write model](#why-not-v2-write-model)
+- [Compatibility](#compatibility)
+- [Test Design](#test-design)
+  * [Functional Tests](#functional-tests)
+  * [Compatibility Tests](#compatibility-tests)
+- [References](#references)
+
+## Introduction
+
+New feature: TiSpark Insert SQL Feature
+
+## Motivation or Background
+
+TiSpark is a connector that provides read, insert and delete support.
+But it only supports insert using DataSource API, users can't use Spark SQL to insert data.
+
+It's common for users to use SQL statements in the big-data workflow. We are going to support TiSpark to
+write using Spark SQL.
+
+## Goals
+**Must have**
+- Support insert data by Spark SQL
+- Compatible with Spark 3.x
+
+**Nice to have**
+- Type convert compatible with the previous version
+- User doc
+
+## API
+Use spark SQL to insert data in TiSpark.
+```
+// insert with spark sql
+spark.sql("insert into tidb_db.db.table values(x, x...)(y, y...)...")
```

Comment:
Is `tidb_db` necessary here? And why it is `tidb_db`?

Reply:
In fast, I just want to say it's a name of database.